microcosm_pubsub/consumer.py

```diff
@@ -72,7 +73,11 @@ def parse_content(cls, consumer, message):
         Parse and validate the message.
 
         """
-        base_message = consumer.pubsub_message_codecs["_"].decode(message)
+        try:
```

Comment:
I think my plan would be:

1. Change codec.py to not have a `default` codec and to remove the `_` dict mappings key. It should be opt-in to set a default codec so that it is an error to not configure a codec for a media type. The use of `defaultdict` would need to be conditional on having a `default` value configured.

2. The consumer should have an "envelope decoding" function; this should default to the current behavior, using the `MediaType` schema; it should be possible to wire in an alternate decoding function by setting an appropriate object graph key.

 3. The simple schema and CLI produce/consume should probably be deleted or moved to an example directory (and removed from the README); they only add confusion because people expect them to work with custom schemas and they don't.